planner/core/common_plans.go

```diff
@@ -454,6 +456,8 @@ type Update struct {
 
 	OrderedList []*expression.Assignment
 
+	NotAllConstant bool
```

Comment:
It's not easy to understand the meaning of `!NotAllConstant`. How about renaming it to `AllAssignmentsAreConstant`